h2o-py/h2o/estimators/pca.py

```diff
@@ -268,4 +268,16 @@ def max_runtime_secs(self, max_runtime_secs):
         assert_is_type(max_runtime_secs, None, numeric)
         self._parms["max_runtime_secs"] = max_runtime_secs
 
+    def transform(self, X, y=None, **params):
```

Comment:
It is not saved to have the property named the same as this method. This method is used in `sklearn.pipeline` and cannot be refactored. How important is to have this property in H2OPrincipalComponentAnalysisEstimator? https://github.com/h2oai/h2o-3/blob/682fdfcc7375e2dfab15b45c512491ecd21985d7/h2o-py/h2o/estimators/pca.py#L117

Reply:
Not sure if I understand. What do you suggest to do instead?

Reply:
I am not a Python guru, but I think there should not be two methods with the same name and absolutely different functionality. There are property `transform(self)` which return value from `parms` dictionary; and now also new method `transform(self, X, y=None, **params)` which calculates something. Before this change, the problem was `sklearn.Pipeline` at one moment call `transform(X)` function on estimator and the property `transform(self)` method returned None (or string with a different setting) instead of a frame... So my question is, how important is to have `transform(self)` property called `transform`? For example, the better name for this property is `transformation` or something like this. I am not sure, how big impact this refactoring will have on usage PCA estimator.

Reply:
I see. This is unfortunate. I think we should deprecate the transform parameter and introduce a new name instead. We will have to keep the old one for a little bit for backwards compatibility. Can you please make a jira issue?

Otherwise if everything is working - I would tolerate the conflict for a while. But we do need to address it eventually.

Reply:
It seems to be working, but it doesn't. If the user set the `transform` property after creating estimator and before using in a pipeline, it will cause the same error.

```python
pca = H2OPCA(k=2)
pca.transform = None
pipe = Pipeline([("standardize", H2OScaler()),
                     ("pca", pca),
                     ("rf", H2ORandomForestEstimator(seed=42,ntrees=50))])
pipe.fit(iris[:4],iris[4])`
```